org.scala-refactoring.library/src/main/scala/scala/tools/refactoring/common/tracing.scala

```diff
@@ -21,7 +26,32 @@ trait Tracing {
   def trace(msg: => String)
 }
 
-trait ConsoleTracing extends Tracing {
+object DebugTracing {
+  private val debugStream = {
+    Option(System.getProperty("scala.refactoring.traceFile")).flatMap { fn =>
```

Comment:
I can move
```scala
  private def printLine(str: String) = {
    debugStream.println(str)
  }
```
into `trait DebugTracing` where clients could easily override it with an implementation that writes to an arbitrary logger.

As for the property `scala.refactoring.traceFile`: I can add this as a constant to `DebugTracing`. Tell me if you know a better place, but note that this really just an option for people debugging the library and not meant for public consumption.